airflow/jobs/scheduler_job_runner.py

```diff
@@ -277,9 +288,51 @@ def __get_concurrency_maps(self, states: Iterable[TaskInstanceState], session: S
             .where(TI.state.in_(states))
             .group_by(TI.task_id, TI.run_id, TI.dag_id)
         )
-        return ConcurrencyMap.from_concurrency_map(
-            {(dag_id, run_id, task_id): count for task_id, run_id, dag_id, count in ti_concurrency_query}
+        ti_concurrency = {
+            (dag_id, run_id, task_id): count for task_id, run_id, dag_id, count in ti_concurrency_query
+        }
+
+        tg_concurrency_query: Iterator[tuple[str, str, str, int]] = session.execute(
+            select(TI.task_id, TI.run_id, TI.dag_id, TI.map_index)
+            .where(
+                TI.map_index >= 0,
+                TI.state.in_(
+                    (
+                        TaskInstanceState.SCHEDULED,
+                        TaskInstanceState.QUEUED,
+                        TaskInstanceState.RUNNING,
+                        TaskInstanceState.UP_FOR_RESCHEDULE,
+                        TaskInstanceState.UP_FOR_RETRY,
+                    )
```

Comment:
Can this use `State.unfinished`, or are there specific reasons some of those states cannot be considered here?

Reply:
`Sate.unfinished` has three more status: 
- TaskInstanceState.SHUTDOWN,
- TaskInstanceState.RESTARTING,
- TaskInstanceState.DEFERRED,
I'm not sure about their usage.
Let me check.

Reply:
SHUTDOWN will go FAILED, which is deprecated.
RESTARTING / DEFERRED will be retried.
It's fine to use `State.unfinished`.

Let me change the code.

Reply:
None means the task instance has not been touched by the scheduler at all, which is impossible in this context. So I would suggest using `TaskInstance.state.in_(s for s in State.unfinished if s is not None)`